modules/navier_stokes/src/executioners/SIMPLE.C

```diff
@@ -0,0 +1,1106 @@
+  {
+    const auto & energy_petsc_options = getParam<MultiMooseEnum>("energy_petsc_options");
+    const auto & energy_petsc_pair_options = getParam<MooseEnumItem, std::string>(
+        "energy_petsc_options_iname", "energy_petsc_options_value");
+    Moose::PetscSupport::processPetscFlags(energy_petsc_options, _energy_petsc_options);
+    Moose::PetscSupport::processPetscPairs(
+        energy_petsc_pair_options, _problem.mesh().dimension(), _energy_petsc_options);
+
+    _energy_ls_control.real_valued_data["rel_tol"] = getParam<Real>("energy_l_tol");
+    _energy_ls_control.real_valued_data["abs_tol"] = getParam<Real>("energy_l_abs_tol");
+    _energy_ls_control.int_valued_data["max_its"] = getParam<unsigned int>("energy_l_max_its");
+
+    // We only allow the solve for a solid energy system if we already solve for the fluid energy
+    if (_has_solid_energy_system)
+    {
+      const auto & solid_energy_petsc_options =
+          getParam<MultiMooseEnum>("solid_energy_petsc_options");
+      const auto & solid_energy_petsc_pair_options = getParam<MooseEnumItem, std::string>(
+          "solid_energy_petsc_options_iname", "solid_energy_petsc_options_value");
+      Moose::PetscSupport::processPetscFlags(solid_energy_petsc_options,
+                                             _solid_energy_petsc_options);
+      Moose::PetscSupport::processPetscPairs(solid_energy_petsc_pair_options,
+                                             _problem.mesh().dimension(),
+                                             _solid_energy_petsc_options);
+
+      _solid_energy_ls_control.real_valued_data["rel_tol"] = getParam<Real>("solid_energy_l_tol");
+      _solid_energy_ls_control.real_valued_data["abs_tol"] =
+          getParam<Real>("solid_energy_l_abs_tol");
+      _solid_energy_ls_control.int_valued_data["max_its"] =
+          getParam<unsigned int>("solid_energy_l_max_its");
+    }
+  }
+
+  if (_has_passive_scalar_systems)
+  {
+    const auto & passive_scalar_petsc_options =
+        getParam<MultiMooseEnum>("passive_scalar_petsc_options");
+    const auto & passive_scalar_petsc_pair_options = getParam<MooseEnumItem, std::string>(
+        "passive_scalar_petsc_options_iname", "passive_scalar_petsc_options_value");
+    Moose::PetscSupport::processPetscFlags(passive_scalar_petsc_options,
+                                           _passive_scalar_petsc_options);
+    Moose::PetscSupport::processPetscPairs(passive_scalar_petsc_pair_options,
+                                           _problem.mesh().dimension(),
+                                           _passive_scalar_petsc_options);
+
+    _passive_scalar_ls_control.real_valued_data["rel_tol"] = getParam<Real>("passive_scalar_l_tol");
+    _passive_scalar_ls_control.real_valued_data["abs_tol"] =
+        getParam<Real>("passive_scalar_l_abs_tol");
+    _passive_scalar_ls_control.int_valued_data["max_its"] =
+        getParam<unsigned int>("passive_scalar_l_max_its");
+  }
+
+  _time = 0;
```

Comment:
need some errors for machinery that isnt done in this executioner:
- adaptivity
- multiapps

Reply:
Multiapps should work (add the exec on final) to the executioner. Added error for adaptivity.

Reply:
it will work for multiapps set to execute on final, but will ignore timestep_begin, end, fixed_point, etc

as you want between an error and adding all the calls

Reply:
Added errors!